### PR TITLE
Ports Guillotine

### PR DESCRIPTION
**Pull request by Fox-McCloud**
Ports Guillotines from TG: https://github.com/tgstation/tgstation/pull/36540

Guillotines are...well, I don't think they really need explanation. They're perfect for Revolutionary activity and executing those pesky Heads of Staff.

Drag someone to one, then buckle them in; pull the lever with harm intent and it'll lop off their head.

The blade becomes dull, over time, so you'll have to eventually sharpen it with a whestone.

Can be constructed with wood, plasteel, and cable coil.

There's a few minor differences between this and TG's version; this uses sounds ever so differently and the layering for when you're buckled to it is different (our sprite system for human mobs doesn't support what TG does with this). 

img

Lastly, clapping will now actually play a sound; what's an ovation without uproarious applause?

:cl: Fox McCloud
add: Adds Guillotines: make them with plasteel, wood, and cable coil. Execute your comdom head of staff TODAY
add: Clapping now actually plays a clapping sound
/:cl:

### CODE REVIEW

**Comment by KasparoVy** [comment]
Mfw we get guillotines but no nooses what the hell

File: code/game/objects/structures/guillotine.dm [hunk 1]
```diff
@@ -0,0 +1,229 @@
...
+
+			// The crowd is pleased
+			// The delay is to make large crowds have a longer lasting applause
+			var/delay_offset = 0
+			for(var/mob/living/carbon/human/HM in viewers(src, 7))
+				addtimer(HM, "emote", delay_offset * 0.3, FALSE, "clap")
+				delay_offset++
+		else
+			H.apply_damage(15 * blade_sharpness, BRUTE, head)
+			add_attack_logs(user, H, "dropped the blade on non-fatally with [src]")
```
Comment (FalseIncarnate) [inline code/game/objects/structures/guillotine.dm]: Probably should re-order these words to "non-fatally dropped the blade on" or something to that effect. Current wording is clunky as hell

**Comment by alexkar598** [comment]
@Fox-McCloud why do i have the feeling this is paving the way for rev being putted back in rotation?

**Comment by shazbot194** [comment]
I'm not saying I dislike this, but I really have to ask why?

**Comment by HereticChurch** [comment]
You're actually doing it you absolute madman.

**Comment by Fox-McCloud** [comment]
Nooses were awful code that required snowflake in multiple places just to work; this is entirely, 100% self-contained.

**Comment by Spartan6** [comment]
Changeling Exterminator 9000

**Comment by granodd** [comment]
Alternatively, construct it in maintenance -behind a fake wall or just out of sight- stunprod someone and then drag them to it to quickly decap them, either to just kill them, take their shit, or steal their head/brain.

**Comment by Fox-McCloud** [comment]
`then drag them to it to quickly decap them, either to just kill them, take their shit, or steal their head/brain`

If they can build a Guillotine, and keep you still long enough to decap you, then they can build a surgery table and decap you with a saw/hatchet.

**Comment by granodd** [comment]
Provided, it takes fifteen seconds to build the guillotine itself. But after the guillotine is built, it only takes three seconds for the guillotine to actually decapitate. Meanwhile, getting someone to hold still for long enough is easy with cuffs of any kind. The fifteen seconds that it takes you to build said guillotine doesn't really matter all that much either if you're not on the radar yet, or are out of sight wherever you do it.

Looking through that code, I didn't see anything relating to a delay on buckling someone into the guillotine. So with that, I'd have to assume that it's the same speed as buckling someone into a bed or chair - that is, instant.

Decapitating someone with a circular saw on a surgery table takes ten seconds, and if you're doing it with a hatchet, there's a chance of failure, therefore potentially making it take longer. The advantage a guillotine would present over a surgery table, despite the resources it takes to build (which aren't really that difficult to come by if you know where to look) is that it takes a significantly shorter amount of time to do its job. Someone being dragged to a surgery table for decap has more of a chance at rescue than someone being dragged to a guillotine. Granted, a surgery table sitting in the middle of maintenance is a bit more innocuous than a guillotine.

tl; dr: the difference between a guillotine and a surgery table is that the guillotine takes a bit more resources and time to build, but does its job faster (3 seconds to decapitate), while the surgery table takes fewer resources, but longer to serve your end of decapitation (10 seconds).

Bonus round: assuming each decapitation takes only 1 sharpness away from the guillotine's blade, you could in theory decapitate three people with the thing before it went dull, provided you had the robustness in your veins.

**Comment by KasparoVy** [comment]
Doesn't matter, decapitated

**Comment by bigfatbananacyclops** [comment]
clearly i suggested this too, yes please

**Comment by FreeStylaLT** [comment]
the gas chamber in brig should just have this instead

**Comment by KasparoVy** [comment]
Gas em and take their head off at the same time, you savages.

**Comment by Fox-McCloud** [comment]
@KasparoVy 

Why not put their corpse into the electric chair after that?

**Comment by KasparoVy** [comment]
:thinking: You absolute genius

**Comment by theColdflame** [comment]
this seems like it's got genuine combat usage rather than being a fluff thing, easy as shit to build and an instant decap

just something to consider

**Comment by Vivalas** [comment]
>maintainers use MRP as an excuse to close PRs
>adds guillotine

**Comment by KasparoVy** [comment]
It's a meme

**Comment by Fox-McCloud** [comment]
Whoah whoah, let's not lose our heads over this.

**Comment by granodd** [comment]
@Fox-McCloud just a heads up (no pun intended), using the guillotine on someone without a head seems to break it. As in, the blade can't be raised back up, and the person inside can't be unbuckled (unless that person is somehow conscious, in which case resisting should set them free if they're unrestrained).

Also the fact that this thing can be wrenched and unwrenched to be moved around has some very, very silly implications...

**Comment by Fox-McCloud** [comment]
I'd rather not mess with and touch human icons; they're an absolute mess.

**Comment by taukausanake** [comment]
Does it kill IPCs if they are put in it? It's supposed to be an execution machine but it shouldn't work on IPCs due to being able to live without a head

**Comment by granodd** [comment]
@taukausanake I tested it and it doesn't. It's actually kind of funny because the guillotine will decap the IPC and then proc them to start clapping, almost as if they were unamused.

I personally don't like that it forces you to clap, because it makes everyone around it clap, even when their character would have a disposition not to. Maybe if they were revolutionaries, though I don't know how deep down the rabbit hole that goes code wise.

**Comment by KasparoVy** [comment]
Lol, it'd be pretty easy to check to see if they're revs but then there's the argument that it could then be used to detect antagonists. Additionally, it would be a snowflake check.

**Comment by shazbot194** [comment]
It forcing you to clap is in quite poor taste. Lets say the captain was doing an execution on the bridge, and a massive mob showed up to protest and break in to the bridge if he does. And as this dramatic scene reaches a crescendo, with the guillotine falling, and everyone protesting starts to clap as if it was a good show. That doesn't sound reasonable to me.

Much like @Vivalas said, this seems very low RP without revs, and other PRs have been shut down for that.

**Comment by tigercat2000** [comment]
I don't really care for the auto-emote either...

**Comment by granodd** [comment]
Merged already? Hmmm.
